### PR TITLE
Documentation link fix

### PR DESCRIPTION
**Pull request by kyzh**
Third time  I try to do that pull request :)
Hopefully this one will be ok.
Bellow original explanation on why we need these 2 commits:

While i was doing a new install, i realised that all link on the default pages were broken.
This is an attempt at fixing it.

I replaced the link to their equivalent on the manual.
So far the version is hardcoded to 1.1, that might need to be change in the long run.
I also replaced the "wiki" link at the bottom of the page by "manual" as i think the wiki is to be deprecated.

Also i just updated the following with the new URL to the "issues" page :
- app/models/orchestration/dhcp.rb
- app/views/common/500.rhtml

Also spotted few places where URL could do with an update:

```
extras/cli/foremancli:puts "     http://theforeman.org/projects/foreman/wiki/Search_API"
extras/cli/foremancli:opts.on( "--custom COLLECTION", "Custom COLLECTION string, see: http://theforeman.org/projects/foreman/wiki/API" ) do |f|
extras/foreman-release.spec:URL: http://theforeman.org/projects/foreman/
```

### CODE REVIEW

File: app/models/host_mailer.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ def summary(options = {})
     filter = []
 
     if (@url = Setting[:foreman_url]).empty?
-      raise ::Foreman::Exception.new N_("':foreman_url:' entry in Foreman configuration file, see http://theforeman.org/projects/foreman/wiki/Mail_Notifications")
+      raise ::Foreman::Exception.new N_("':foreman_url:' entry in Foreman configuration file, see http://www.theforeman.org/manuals/1.1/index.html#3.5.4PuppetReports")
```
Comment (domcleal) [inline app/models/host_mailer.rb]: We'd best leave this for now, as mail notifications aren't covered in the manual.

File: app/views/common/500.rhtml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<% title "Oops, we're sorry but something went wrong" %>
+
+<div class="alert alert-message alert-block alert-error base in fade">
+  <a class="close" href="#" data-dismiss="alert">x</a>
+    <%= h exception.message %>
+</div>
+
+<p>
+  If you feel this is an error with Foreman itself, please open a new issue with <%= link_to "Foreman ticketing system", "http://projects.theforeman.org/projects/foreman/issues", :rel => "external" %>,
```
Comment (domcleal) [inline app/views/common/500.rhtml]: The filename's changed here, you'll need to change 500.html.erb instead.

File: app/views/puppetclasses/no_route.html.erb [hunk 3]
```diff
@@ -4,7 +4,7 @@
 <div class="alert alert-message alert-block alert-error">
   <h3><%= _('Oops') %></h3>
   <p>
-  <%= _('Either you didn''t generate the puppetdocs (see <a rel="external" href="%{url}">here</a>) or the class <strong>%{name}</strong> could not be found in environment <strong>%{environment}</strong><br/> Please ensure that the class <strong>%{name}</strong>''s declaration is accessible via the modulepath associated with <strong>%{environment}</strong>.') % { :name => name, :environment => environment, :url => 'http://theforeman.org/projects/foreman/wiki/Puppet_class_browser' } %>
+  <%= _('Either you didn''t generate the puppetdocs (see <a rel="external" href="%{url}">here</a>) or the class <strong>%{name}</strong> could not be found in environment <strong>%{environment}</strong><br/> Please ensure that the class <strong>%{name}</strong>''s declaration is accessible via the modulepath associated with <strong>%{environment}</strong>.') % { :name => name, :environment => environment, :url => 'http://www.theforeman.org/manuals/1.1/index.html#3.5.2ConfigurationOptions' } %>
```
Comment (domcleal) [inline app/views/puppetclasses/no_route.html.erb]: Probably need to leave this too, it hasn't been moved.

File: app/views/reports/welcome.html.erb [hunk 4]
```diff
@@ -2,9 +2,9 @@
 <div id="welcome">
   <p>
   <%= _("You don't seem to have any reports, if you wish to configure Puppet to forward it reports to Foreman, please follow") %>
-  <%= link_to _("setting up reporting"), "http://theforeman.org/projects/foreman/wiki/Puppet_Reports", :rel => "external" %>
+  <%= link_to _("setting up reporting"), "http://theforeman.org/manuals/1.1/index.html#3.5.2ConfigurationOptions", :rel => "external" %>
```
Comment (domcleal) [inline app/views/reports/welcome.html.erb]: This should be the URL below, `http://www.theforeman.org/manuals/1.1/index.html#3.5.4PuppetReports`

File: app/views/reports/welcome.html.erb [hunk 5]
```diff
@@ -2,9 +2,9 @@
...
   <%= _("and") %>
-  <%= link_to _("e-mail reporting"), "http://theforeman.org/projects/foreman/wiki/Mail_Notifications", :rel => "external" %>.
+  <%= link_to _("e-mail reporting"), "theforeman.org/manuals/1.1/index.html#3.5.4PuppetReports", :rel => "external" %>.
```
Comment (domcleal) [inline app/views/reports/welcome.html.erb]: Best leave this again.

**Comment by domcleal** [comment]
Thanks @kyzh, this is great - a few comments added above.  As per [the ticket](http://projects.theforeman.org/issues/2167), using the version string would be a good improvement.  Just force push to the branch and the PR will update.

**Comment by kyzh** [comment]
as far as i can tell 842a721 address all the above concern you had.

I have to admit that i didn't test 7120dcc yet and it is probably why it is failling, will give it a try later today.
I also need to understand how to change it within `link_to`

File: app/views/auth_source_ldaps/_form.html.erb [hunk 6]
```diff
@@ -1,4 +1,4 @@
-<% title_actions link_to(icon_text("question-sign", _("Documentation"), :class => "icon-white"), "http://theforeman.org/projects/foreman/wiki/LDAP_Authentication", :rel => "external", :class => "btn-info") %>
+<% title_actions link_to(icon_text("question-sign", _("Documentation"), :class => "icon-white"), "http://www.theforeman.org/manuals/1.1/index.html#4.1.1LDAPAuthentication", :rel => "external", :class => "btn-info") %>
```
Comment (ohadlevy) [inline app/views/auth_source_ldaps/_form.html.erb:1]: can you change these to something like:

``` diff
-<% title_actions link_to(icon_text("question-sign", _("Documentation"), :class => "icon-white"), "http://theforeman.org/projects/foreman/wiki/LDAP_Authentication", :rel => "external", :class => "btn-info") %>
+<% title_actions link_to(icon_text("question-sign", _("Documentation"), :class => "icon-white"), "http://www.theforeman.org/manuals/#{SETTINGS[:version] %>/index.html#4.1.1LDAPAuthentication", :rel => "external", :class => "btn-info") %>
```

File: app/views/common/500.html.erb [hunk 7]
```diff
@@ -1,4 +1,4 @@
-<% title _("Oops, we're sorry but something went wrong") %>
+<% title "Oops, we're sorry but something went wrong" %>
```
Comment (ohadlevy) [inline app/views/common/500.html.erb:1]: why revert this?

File: app/views/common/500.html.erb [hunk 8]
```diff
@@ -6,14 +6,14 @@
 </div>
 
 <p>
-  <%= _("If you feel this is an error with Foreman itself, please open a new issue with") %> <%= link_to _("Foreman ticketing system"), "http://theforeman.org/projects/foreman/issues", :rel => "external" %>,
-  <%= _("You would probably need to attach the") %>
-  <%= link_to_function(_("Full trace"), "$('#backtrace').toggle()")%> <%= _("and relevant log entries.") %>
+  If you feel this is an error with Foreman itself, please open a new issue with <%= link_to "Foreman ticketing system", "http://projects.theforeman.org/projects/foreman/issues", :rel => "external" %>,
+  You would probably need to attach the
+  <%= link_to_function "Full trace", toggle_div("backtrace") %> and relevant log entries.
```
Comment (ohadlevy) [inline app/views/common/500.html.erb:11]: this seems like you removed the string extractions? this is used for translations.

File: app/views/auth_source_ldaps/_form.html.erb (hunk 6 above)
Comment (domcleal) [inline app/views/auth_source_ldaps/_form.html.erb:1]: I don't think that's a good idea as :version might contain sub-versions - e.g. we currently have 1.1-1, which isn't on the web site.

Comment (kyzh) [inline app/views/auth_source_ldaps/_form.html.erb:1]: I'm a bit lost now.
I see 4 way to go about it:
1.  Use only major.minor version.
2.  Have different file for major/minor version and revision version.
3.  Create a version of the documentation per revision version.
4.  Change the behavior server side.

1) I could put a regex which doesn't feel right, but would work
2) I'm not sure what impact and what kind of headache that will cause later on
3) I don't know what are the plan for documentation versioning , if the 1.2 will be the same than 1.2.25 or not etc..
4) What about redirecting `x.y.z` to `x.y` on the server side and ask ourself this question when an actual problem appears?

File: app/views/common/500.html.erb (hunk 7 above)
Comment (kyzh) [inline app/views/common/500.html.erb:1]: Bad use of git/merge on my sidesorry

File: app/views/common/500.html.erb (hunk 8 above)
Comment (kyzh) [inline app/views/common/500.html.erb:11]: Bad use of git/merge (again) on my sidesorry

**Comment by kyzh** [comment]
I will try to push the changes, but i'm still not sure on what approch you would like me to take.
i think i will do something  like `SETTINGS[:version].split("-")[0]` 
if that doesn't cut it still, we can change it then

**Comment by lzap** [comment]
Lemme fix that now.

**Comment by domcleal** [comment]
@lzap's submitted #774 with some fixes, closing this one.  Thanks @kyzh!

**Pull request closed without merging**